buildSrc/build.gradle.kts

```diff
@@ -17,17 +15,6 @@ dependencies {
   implementation(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
 }
 
-java {
-  sourceCompatibility = JavaVersion.VERSION_11
-  targetCompatibility = JavaVersion.VERSION_11
-}
-
 kotlin {
-  target {
-    compilations.configureEach {
-      kotlinOptions {
-        jvmTarget = "11"
-      }
-    }
-  }
+  jvmToolchain(11)
```

Comment:
I think it would it make sense to configure the global toolchain here as well, for consistency, via the `java { toolchain {  } }` block.

Reply:
The `build.gradle.kts` is only responsible for the code within `buildSrc`, right?
Since we are using Kotlin only here, why should we set the `java.toolchain {}`? 🤔 

Or what do you mean by that?

Reply:
Yes, you are correct that this build script is responsible for `buildSrc`.

In Gradle, the `java` plugin is _always_ applied to JVM-based projects, even if the project uses some other language as the core of its implementation. Therefore, for consistency, it is considered idiomatic to use the `java` plugin methods to configure something first, and use language-specific methods to adjust the configuration, if necessary.

In the toolchain case, it is specified explicitly in Kotlin docs (see [here](https://kotlinlang.org/docs/gradle-configure-project.html#gradle-java-toolchains-support)) that the toolchain version in the `java` block will be used by Kotlin compilation tasks, therefore, it would be conventional to use the `java` block to set it up.

(BTW, it's curious that the docs state that setting the toolchain in the `kotlin` extension will also configure it for `JavaCompile` tasks; this is another reason to set it in the `java` block, since it is the built-in way to configure the toolchain, and their effects would be similar anyway)